pkg/proxy/healthcheck/proxier_health.go

```diff
@@ -109,18 +120,42 @@ type healthzHandler struct {
 }
 
 func (h healthzHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
-	lastUpdated := time.Time{}
+	var lastQueued, lastUpdated time.Time
+	if val := h.hs.lastQueued.Load(); val != nil {
+		lastQueued = val.(time.Time)
+	}
 	if val := h.hs.lastUpdated.Load(); val != nil {
 		lastUpdated = val.(time.Time)
 	}
 	currentTime := h.hs.clock.Now()
 
+	healthy := false
+	switch {
+	case lastUpdated.IsZero():
+		// Consider the proxy healthy while it's starting up
+		healthy = true
+	case lastUpdated.After(lastQueued):
+		// We've processed all updates
+		healthy = true
+	case currentTime.Sub(lastQueued) < h.hs.healthTimeout:
```

Comment:
I don't have time to spelunk right this minute, but won't the change-tracking logic elide an A-B-A change, such that this would believe a change was queued but sync() doesn't actually need to run?  e.g. change port number and then change it back.

Reply:
If we elide a change then `lastQueued` won't get updated. We call `proxier.healthzServer.QueuedUpdate()` immediately before calling `proxier.syncRunner.Run()`.